kinto/core/resource/viewset.py

```diff
@@ -118,14 +128,17 @@ def get_view_arguments(self, endpoint_type, resource_cls, method):
         endpoint_args = getattr(self, by_method, {})
         args.update(**endpoint_args)
 
+        request_schema = args.get('schema', RequestSchema)
+
+        # Instantiate request schema
+        request_schema = request_schema()
+
         if method.lower() in map(str.lower, self.validate_schema_for):
-            schema = RequestSchema()
             record_schema = self.get_record_schema(resource_cls, method)
             record_schema.name = 'body'
-            schema.add(record_schema)
-            args['schema'] = schema
-        else:
-            args['schema'] = RequestSchema()
+            request_schema.add(record_schema)
```

Comment:
An interesting approach IMO would be to handle the `record_schema` binding directly on the schema (instead of the viewset), what do you think?